src/ctap/mod.rs

```diff
@@ -427,22 +442,29 @@ where
             return Err(Ctap2StatusCode::CTAP2_ERR_UNSUPPORTED_ALGORITHM);
         }
 
-        let (use_hmac_extension, cred_protect_policy) = if let Some(extensions) = extensions {
-            let mut cred_protect = extensions.cred_protect;
-            if cred_protect.unwrap_or(CredentialProtectionPolicy::UserVerificationOptional)
-                < DEFAULT_CRED_PROTECT
-                    .unwrap_or(CredentialProtectionPolicy::UserVerificationOptional)
-            {
-                cred_protect = DEFAULT_CRED_PROTECT;
-            }
-            (extensions.hmac_secret, cred_protect)
-        } else {
-            (false, DEFAULT_CRED_PROTECT)
-        };
+        let rp_id = rp.rp_id;
+        let (use_hmac_extension, cred_protect_policy, min_pin_length) =
+            if let Some(extensions) = extensions {
+                let mut cred_protect = extensions.cred_protect;
+                if cred_protect.unwrap_or(CredentialProtectionPolicy::UserVerificationOptional)
```

Comment:
This code block seems equivalent to
```rust
let cred_protect = Some(cmp::max(
  extensions.cred_protect.unwrap_or(CredentialProtectionPolicy::UserVerificationOptional),
  DEFAULT_CRED_PROTECT.unwrap_or(CredentialProtectionPolicy::UserVerificationOptional)
));
```

Reply:
There is a slight difference. Practically, `Some(CredentialProtectionPolicy::UserVerificationOptional)` and `None` should be equivalent, but I need to support both for backwards compatibility. And I rather return `None` in case the platform doesn't speak `credProtect`. Because having `Some` policy means we return the extension output. I don't think the current version is elegant, but I didn't find a better alternative so far.

Reply:
It's just the way Github takes code snippet that is confusing here I think.
The equivalent code block I referred to is:
```rust
let mut cred_protect = extensions.cred_protect;
if cred_protect.unwrap_or(CredentialProtectionPolicy::UserVerificationOptional)
    < DEFAULT_CRED_PROTECT
        .unwrap_or(CredentialProtectionPolicy::UserVerificationOptional)
{
    cred_protect = DEFAULT_CRED_PROTECT;
}
```

which should be a perfect equivalent of:
```rust
let cred_protect = Some(cmp::max(
  extensions.cred_protect.unwrap_or(CredentialProtectionPolicy::UserVerificationOptional),
  DEFAULT_CRED_PROTECT.unwrap_or(CredentialProtectionPolicy::UserVerificationOptional)
));
```

Reply:
This is what I thought you mean, and my comment applies. I want this piece of code to always return `None` instead of `Some(CredentialProtectionPolicy::UserVerificationOptional)`.